addons/mail/static/src/js/form_renderer.js

```diff
@@ -44,25 +44,27 @@ FormRenderer.include({
     //--------------------------------------------------------------------------
 
     /**
-     * Overrides the function that renders the nodes to return the chatter's $el
-     * for the 'oe_chatter' div node.
-     * Returns an empty div instead of the chatter's $el in create mode.
+     * Overrides the function that renders the nodes to process the 'oe_chatter'
+     * div node: we instantiate (or update if it already exists) the chatter,
+     * and we return a fake node that we will use as a hook to insert the
+     * chatter into the DOM when the whole view will be rendered.
+     * In create mode returns an empty div instead of the fake hook node.
      *
      * @override
      * @private
      */
     _renderNode: function (node) {
         if (node.tag === 'div' && node.attrs.class === 'oe_chatter') {
-            if (this.chatter) {
-                // Detach the chatter before updating the $el.
-                // This is important because if the view is now in create mode
-                // (edit mode with no res_id), the chatter will be removed from
-                // the DOM, and its handlers will be unbound. By detaching it
-                // beforehand, we ensure to keep its handlers alive so that if
-                // it is re-appended later, everything will still work properly
-                this.chatter.$el.detach();
-            }
             if (this.mode === 'edit' && !this.state.data.id) {
+                if (this.chatter) {
+                    // Detach the chatter before updating the $el.
+                    // This is important because if the view is now in create mode
+                    // (edit mode with no res_id), the chatter will be removed from
+                    // the DOM, and its handlers will be unbound. By detaching it
+                    // beforehand, we ensure to keep its handlers alive so that if
+                    // it is re-appended later, everything will still work properly
+                    this.chatter.$el.detach();
+                }
```

Comment:
you should never detach the here, as this will cause a flickering

Reply:
@aab-odoo Flickering does not occur in Create case, I tried but I did not face flickering issue, but still I improved code as we want to detach chatter element in create/edit mode and we will replace temp hook with chatter element only if record has ID, as in create mode we will not have temp hook.